cellrank/external/kernels/_statot_kernel.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, Union, Optional
+
+from typing_extensions import Literal
+
+from anndata import AnnData
+
+import numpy as np
+import pandas as pd
+
+from cellrank.ul._docs import d
+from cellrank.tl.estimators import GPCCA
+from cellrank.tl.kernels._precomputed_kernel import DummyKernel
+
+_error = None
+try:
+    from statot import OTKernel as OTKernel_
+except ImportError as e:
+    from cellrank.external.kernels._import_error_kernel import (
+        ErroredKernel as OTKernel_,
+    )
+
+    _error = e
+
+
+@d.dedent
+class OTKernel(OTKernel_, error=_error):
+    """
+    Stationary optimal transport kernel from [Zhang21]_.
+
+    This class requires the `statOT` package, which can be installed as `pip install statot POT`.
+
+    Parameters
+    ----------
+    %(adata)s
+    terminal_states
+        Categorical :class:`pandas.Series` where non-`NaN` values mark terminal states.
+        If `None`, terminal states are assumed to be already present in :paramref:`adata` `['terminal_states']`.
+    g
+        Key in :attr:`anndata.AnnData.obs` containing relative growth rates for cells or the array itself.
+    cluster_key
+        If a key to cluster labels is given, `terminal_states` will be associated with these for naming and colors.
+    kwargs
+        Additional keyword arguments.
+    """
+
+    __import_error_message__ = "Unable to import the kernel. Please install `statOT` first as `pip install statot POT`."
+
+    def __init__(
+        self,
+        adata: AnnData,
+        g: Union[str, np.ndarray],
+        terminal_states: Optional[Union[str, pd.Series]] = None,
+        cluster_key: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        if terminal_states is not None:
```

Comment:
@zsteve 

We've converged on this solution (is the cleanest/easiest) - you can still optionally pass `terminal_states` in the `__init__` method (by default it's `None`, i.e. it's assumed they are already present).

Let me know what you think.

Reply:
Yes that sounds fine! So it is agnostic of how the terminal states are assigned. Makes sense!